java/ql/src/utils/model-generator/CaptureSourceModels.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Capture source models.
+ * @description Finds APIs that act as sources as they expose already known sources.
+ * @id java/utils/model-generator/sink-models
+ */
+
+import java
+private import Telemetry.ExternalAPI
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.dataflow.ExternalFlow
+private import ModelGeneratorUtils
+private import semmle.code.java.dataflow.internal.FlowSummaryImplSpecific
+private import semmle.code.java.dataflow.internal.FlowSummaryImpl
+private import semmle.code.java.dataflow.internal.DataFlowImplCommon
+private import semmle.code.java.dataflow.internal.DataFlowNodes::Private
+
+class FromSourceConfiguration extends TaintTracking::Configuration {
+  FromSourceConfiguration() { this = "FromSourceConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { sourceNode(source, _) }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(Callable c |
+      sink instanceof ReturnNodeExt and
+      sink.getEnclosingCallable() = c and
+      c.isPublic() and
+      c.fromSource()
+    )
+    or
+    exists(MethodAccess c | sink.asExpr() = c.getAnArgument())
+  }
+
+  override DataFlow::FlowFeature getAFeature() {
+    result instanceof DataFlow::FeatureHasSinkCallContext
+  }
+}
+
+string asOutput(DataFlow::Node node) {
+  if node instanceof ReturnNodeExt
+  then result = "ReturnValue"
+  else
+    result =
+      "Parameter[" +
+        node.(ArgumentNode)
+            .getCall()
+            .asCall()
+            .getQualifier()
+            .(VarAccess)
+            .getVariable()
+            .(Parameter)
+            .getPosition() + "]"
```

Comment:
Remove this case.

Reply:
Can you elaborate on this? Sure enough, this is far from being complete in terms of supporting lamda flows but it captures some of the more basic usages. Would you want me to do it outside this PR? Not at all?

Here is an example for context
```
| p;Sources;true;consumeSource;(int,Consumer);;Parameter[1];remote |
```

```
  public void consumeSource(int port, Consumer<InputStream> consumer) throws IOException {
      ServerSocket socket = new ServerSocket(port);
      consumer.accept(socket.accept().getInputStream());
  }
```

Reply:
Removed for now.

Reply:
Oh, I had not realised that this was about callbacks. That's definitely something we'll want to postpone as followup work.